src/fmu/dataio/_objectdata_provider.py

```diff
@@ -140,6 +142,52 @@ class TimedataFormat:
     t1: Optional[TimedataValueLabel]
 
 
+@dataclass
+class DerivedNamedStratigraphy:
+    name: str
+    alias: list[str]
+
+    stratigraphic: bool
+    stratigraphic_alias: list[str]
+
+    offset: int | None
+    base: str | None
+    top: str | None
+
+    @staticmethod
+    def from_dop(obj: _ObjectDataProvider) -> DerivedNamedStratigraphy:
+        name = obj.dataio.name
+
+        if not name:
+            try:
+                name = obj.obj.name
+            except AttributeError:
+                name = ""
+
+        # next check if usename has a "truename" and/or aliases from the config
+        strat = obj.dataio.config.get("stratigraphy")  # shortform
+        no_start_or_missing_name = strat is None or name not in strat
+
+        rv = DerivedNamedStratigraphy(
+            name=name if no_start_or_missing_name else strat[name].get("name", name),
+            alias=[] if no_start_or_missing_name else strat[name].get("alias", []),
+            stratigraphic=False
+            if no_start_or_missing_name
+            else strat[name].get("stratigraphic", False),
+            stratigraphic_alias=[]
+            if no_start_or_missing_name
+            else strat[name].get("stratigraphic_alias"),
+            offset=None if no_start_or_missing_name else strat[name].get("offset"),
+            top=None if no_start_or_missing_name else strat[name].get("top"),
+            base=None if no_start_or_missing_name else strat[name].get("base"),
+        )
+
+        if not no_start_or_missing_name and rv.name != "name":
+            rv.alias.append(name)
```

Comment:
Can be moved inside the above function block by usage of walrus op. But gets unreadable.